src/MainPage.jsx

```diff
@@ -280,11 +282,20 @@ class App extends Component {
         return;
       }
     }
-    // Read a card from the customer
-    this.terminal.setSimulatorConfiguration({
+
+    
+
+    const simulatorConfiguration = {
       testPaymentMethod: this.state.testPaymentMethod,
       testCardNumber: this.state.testCardNumber
-    });
+    };
+
+    if (this.state.simulateOnReaderTip) {
+      simulatorConfiguration.tipAmount = Number(this.state.tipAmount);
```

Comment:
When `this.state.simulateOnReaderTip=false` I don't set `tipAmount` to null explicitly here and just omit it because for some reason setting it explicitly to null returns error `parameter_invalid_integer` from Stripe.

IDK why this error happens because [here in docs](https://stripe.com/docs/terminal/references/api/js-sdk#stripeterminal-setsimulatorconfig) it's said that `tipAmount` can be _Any amount or null._

Reply:
got it. How you have it looks great and I'll look in to if we should fix those docs or the simulated reader.